pkg/logs/launchers/integration/launcher_test.go

```diff
@@ -156,6 +156,14 @@ func (suite *LauncherTestSuite) TestDeleteFile() {
 	assert.True(suite.T(), os.IsNotExist(err))
 }
 
+// TestNegativeCombinedUsageMax ensures errors in combinedUsageMax don't result
+// in panics from `deleteFile`
+func (suite *LauncherTestSuite) TestNegativeCombinedUsageMax() {
+	suite.s.combinedUsageMax = -1
```

Comment:
Maybe we can add tests that cover the behavior when: 

`combinedUsageMax` is `0`, `1`, `100` (something large enough to store one or two logs). 
In each case, write several logs from several sources and assert that everything behaves as expected. 
Just to put the components through it's paces and make sure it behaves correctly in anomalous cases.

Reply:
Good idea, adding tests for this now